live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/domain/usecase/e2e/bulk_ingester/BulkIngesterEventIntegrationTest.java

```diff
@@ -0,0 +1,139 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.domain.usecase.e2e.bulk_ingester;
+
+import static org.alfresco.hxi_connector.live_ingester.util.ContainerSupport.REQUEST_ID_PLACEHOLDER;
+
+import org.junit.jupiter.api.Test;
+
+import org.alfresco.hxi_connector.live_ingester.util.E2ETestBase;
+
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+public class BulkIngesterEventIntegrationTest extends E2ETestBase
+{
+    @Test
+    void shouldIngestOnlyMetadataIfThereIsNoContent()
+    {
+        // given
+        containerSupport.prepareHxInsightToReturnSuccess();
+
+        // when
+        String repoEvent = """
+                {
+                  "nodeId": "5018ff83-ec45-4a11-95c4-681761752aa7",
+                  "type": "cm:category",
+                  "createdAt": 1707153552,
+                  "creatorId": "System",
+                  "modifierId": "admin",
+                  "aspectNames": [
+                    "cm:auditable"
+                  ],
+                  "contentInfo": null,
+                  "customProperties": {
+                    "cm:name": "Mexican Spanish"
+                  }
+                }""";
+        containerSupport.raiseBulkIngesterEvent(repoEvent);
+
+        // then
+        String expectedBody = """
+                {
+                   "objectId" : "5018ff83-ec45-4a11-95c4-681761752aa7",
+                   "eventType" : "create",
+                   "properties" : {
+                      "type": "cm:category",
+                      "createdAt": 1707153552,
+                      "createdByUserWithId" : "System",
+                      "modifiedByUserWithId" : "admin",
+                      "aspectsNames" : [ "cm:auditable" ],
+                      "cm:name" : "Mexican Spanish"
+                    }
+                 }""";
+        containerSupport.expectHxIngestMessageReceived(expectedBody);
+    }
+
+    @Test
+    void shouldIngestMetadataAndContent()
+    {
+        // given
+        containerSupport.prepareHxInsightToReturnSuccess();
+
+        // when
+        String repoEvent = """
+                {
+                  "nodeId": "37be157c-741c-4e51-b781-20d36e4e335a",
+                  "type": "cm:content",
+                  "creatorId": "admin",
+                  "modifierId": "hr_user",
+                  "aspectNames": [
+                    "cm:indexControl",
+                    "cm:auditable"
+                  ],
+                  "contentInfo": {
+                    "contentSize": 330,
+                    "encoding": "ISO-8859-1",
+                    "mimetype": "text/xml"
+                  },
```

Comment:
I am wondering if we actually need this content-info data? Maybe I could change it to "hasContent" boolean? wdyt @tpage-alfresco, @krdabrowski?

Reply:
We don't pass this data further to HxI?

Reply:
no, we just need info if node contains content or not - I wonder if we will use it in the future 🤔

Reply:
We could potentially use the mimetype in the future to determine whether a transformation is possible. Currently we're just sending everything for transformation.